File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.cpp
class Solution {
public:
    double findMaxAverage(vector<int>& nums, int k) {
        int n = nums.size();
        double max = numeric_limits<int>::max() * -1;
        int sum = 0;
        for(int i = 0;i<k;i++){
            sum += nums[i];
        }
        max = sum>max ? sum:max;
        for(int i =1;i<=n-k;i++){
            sum+=nums[i+k-1];
            sum-=nums[i-1];
            max = sum > max ? sum:max;
        }
        return max / k;
    }

    
};

File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.java
class Solution {
    public double findMaxAverage(int[] nums, int k) {
        double K = k;
        double M = -10000;
        double total = 0;
        for(int i = 0;i<nums.length-k+1;i++){
            if(i==0){
                for(int j = 0;j<k;j++){
                    total+=nums[j];
                }
                M = Math.max(M,total/K);
            }else{
                
                total-=nums[i-1];
                total+=nums[i+k-1];
                
                M = Math.max(M,total/K);
            }
        }
        return M;
    }
}